src/android/kotlin/RustWebViewClient.kt

```diff
@@ -48,6 +49,7 @@ class RustWebViewClient(context: Context): WebViewClient() {
     }
 
     override fun onPageFinished(view: WebView, url: String) {
+        currentUrl = url
```

Comment:
we should use https://developer.android.com/reference/android/webkit/WebView#getUrl() instead

Reply:
getUrl can only be used on the main thread, and i feel it's a bottleneck to the IPC

Reply:
tried it directly, just crashes :(

Reply:
then let's leave a comment and document why we went with this manual approach